src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,674 @@
+	 */
+	public function state( string $store_namespace, array $state = array() ): array {
+		if ( ! isset( $this->state_data[ $store_namespace ] ) ) {
+			$this->state_data[ $store_namespace ] = array();
+		}
+		if ( is_array( $state ) ) {
+			$this->state_data[ $store_namespace ] = array_replace_recursive(
+				$this->state_data[ $store_namespace ],
+				$state
+			);
+		}
+		return $this->state_data[ $store_namespace ];
+	}
+
+	/**
+	 * Gets and/or sets the configuration of the Interactivity API for a given
+	 * store namespace.
+	 *
+	 * If configuration for that store namespace exists, it merges the new
+	 * provided configuration with the existing one.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $store_namespace The unique store namespace identifier.
+	 * @param array  $config          Optional. The array that will be merged with the existing configuration for the
+	 *                                specified store namespace.
+	 * @return array The configuration for the specified store namespace. This will be the updated configuration if a
+	 *               $config argument was provided.
+	 */
+	public function config( string $store_namespace, array $config = array() ): array {
+		if ( ! isset( $this->config_data[ $store_namespace ] ) ) {
+			$this->config_data[ $store_namespace ] = array();
+		}
+		if ( is_array( $config ) ) {
+			$this->config_data[ $store_namespace ] = array_replace_recursive(
+				$this->config_data[ $store_namespace ],
+				$config
+			);
+		}
+		return $this->config_data[ $store_namespace ];
+	}
+
+	/**
+	 * Prints the serialized client-side interactivity data.
+	 *
+	 * Encodes the config and initial state into JSON and prints them inside a
+	 * script tag of type "application/json". Once in the browser, the state will
+	 * be parsed and used to hydrate the client-side interactivity stores and the
+	 * configuration will be available using a `getConfig` utility.
+	 *
+	 * @since 6.5.0
+	 */
+	public function print_client_interactivity_data() {
+		$store      = array();
+		$has_state  = ! empty( $this->state_data );
+		$has_config = ! empty( $this->config_data );
+
+		if ( $has_state || $has_config ) {
+			if ( $has_config ) {
+				$store['config'] = $this->config_data;
+			}
+			if ( $has_state ) {
+				$store['state'] = $this->state_data;
+			}
+			wp_print_inline_script_tag(
+				wp_json_encode(
+					$store,
+					JSON_HEX_TAG | JSON_HEX_AMP
+				),
+				array(
+					'type' => 'application/json',
+					'id'   => 'wp-interactivity-data',
+				)
+			);
+		}
+	}
+
+	/**
+	 * Registers the `@wordpress/interactivity` script modules.
+	 *
+	 * @since 6.5.0
+	 */
+	public function register_script_modules() {
+		wp_register_script_module(
+			'@wordpress/interactivity',
+			includes_url( 'js/dist/interactivity.min.js' )
+		);
+
+		wp_register_script_module(
+			'@wordpress/interactivity-router',
+			includes_url( 'js/dist/interactivity-router.min.js' ),
+			array( '@wordpress/interactivity' )
+		);
```

Comment:
These should use the non-`min` versions when working on `src` or `SCRIPT_DEBUG` is enabled:

```suggestion
		$dev_suffix = wp_scripts_get_suffix( 'dev' );

		wp_register_script_module(
			'@wordpress/interactivity',
			includes_url( "js/dist/interactivity{$dev_suffix}.js" )
		);

		wp_register_script_module(
			'@wordpress/interactivity-router',
			includes_url( "js/dist/interactivity-router{$dev_suffix}.js" ),
			array( '@wordpress/interactivity' )
		);
```

Reply:
It looks like this is also the case for block-specific module scripts. When `SCRIPT` debug is enabled, I'm seeing `view.min.js` being loaded, not the expected `view.js`.

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/5953/commits/e9cb1fe2579c345fd4ea71eea92da17b4853f607.

Reply:
> It looks like this is also the case for block-specific module scripts. When `SCRIPT` debug is enabled, I'm seeing `view.min.js` being loaded, not the expected `view.js`.

It is the case for scripts registered with `viewScript`, because they were always pointing to the minified version:

```json
{
  "viewScript": "file:./view.min.js"
}
```

For WP 6.5, we are switching to script modules and are going to manually register the script modules so we'll add the `wp_scripts_get_suffix` in `block-library` (Gutenberg).